src/Lock.php

```diff
@@ -136,8 +136,19 @@ public static function showForItem(CommonDBTM $item)
                         continue;
                     }
                     $query['WHERE'][] = $connexity_criteria['WHERE'];
+                } else { //is CommonDBTM (ie : Monitor / DatabaseInstance)
```

Comment:
I think it should be filtered only on `$CFG_GLPI['directconnect_types']`, as only these types should be present in `Computer_Item` table, but `DatabaseInstance` is not listed in this config entry.

Reply:
Because  ```DatabaseInstance``` does not use ```Computer_Item``` 

```sql
MariaDB [10bugfixes_ruless]> explain glpi_databaseinstances;
+-------------------------------+------------------+------+-----+---------+----------------+
| Field                         | Type             | Null | Key | Default | Extra          |
+-------------------------------+------------------+------+-----+---------+----------------+
| id                            | int(10) unsigned | NO   | PRI | NULL    | auto_increment |
| entities_id                   | int(10) unsigned | NO   | MUL | 0       |                |
| is_recursive                  | tinyint(4)       | NO   | MUL | 0       |                |
| name                          | varchar(255)     | NO   | MUL |         |                |
| version                       | varchar(255)     | NO   |     |         |                |
| port                          | varchar(10)      | NO   |     |         |                |
| path                          | varchar(255)     | NO   |     |         |                |
| size                          | int(11)          | NO   |     | 0       |                |
| databaseinstancetypes_id      | int(10) unsigned | NO   | MUL | 0       |                |
| databaseinstancecategories_id | int(10) unsigned | NO   | MUL | 0       |                |
| locations_id                  | int(10) unsigned | NO   | MUL | 0       |                |
| manufacturers_id              | int(10) unsigned | NO   | MUL | 0       |                |
| users_id_tech                 | int(10) unsigned | NO   | MUL | 0       |                |
| groups_id_tech                | int(10) unsigned | NO   | MUL | 0       |                |
| states_id                     | int(10) unsigned | NO   | MUL | 0       |                |
| **itemtype**                      | varchar(100)     | NO   | MUL |         |                |
| **items_id**                      | int(10) unsigned | NO   |     | 0       |                |
| is_onbackup                   | tinyint(4)       | NO   |     | 0       |                |
| is_active                     | tinyint(4)       | NO   | MUL | 0       |                |
| is_deleted                    | tinyint(4)       | NO   | MUL | 0       |                |
| is_helpdesk_visible           | tinyint(4)       | NO   | MUL | 1       |                |
| is_dynamic                    | tinyint(4)       | NO   | MUL | 0       |                |
| autoupdatesystems_id          | int(10) unsigned | NO   | MUL | 0       |                |
| date_creation                 | timestamp        | YES  | MUL | NULL    |                |
| date_mod                      | timestamp        | YES  | MUL | NULL    |                |
| date_lastboot                 | timestamp        | YES  |     | NULL    |                |
| date_lastbackup               | timestamp        | YES  |     | NULL    |                |
| comment                       | text             | YES  |     | NULL    |                |
+-------------------------------+------------------+------+-----+---------+----------------+
```